### ISSUE TITLE
request: display which workspaces have windows

### ISSUE THREAD

**Issue by ingolemo** [conformal/spectrwm#170]
I sometimes forget which workspace I have windows on and it would be really useful to me to have this information displayed in the bar somehow. I can sort of work around this issue using `xdotool`, but that's not a great solution. Ideally, I'd like a new bar format sequence that would have a value something like `1 2 5` (or whatever numbers of workspaces currently have windows).

**Comment by mwm** [comment]
MOD left & right to cycle through the workspaces with windows on them isn't sufficient? Or are you using that with xdotool?

How do you feel about color-coding the list? Workspaces that were currently displayed being in bar_border instead of bar_font_color, and the current workspace being in inverse video?

**Comment by ingolemo** [comment]
My actual use-case for this involves empty workspaces. I would like to be able to see at a glance which number key I can press to move the current window to an empty workspace. Obviously, It would be more direct for the bar to show which workspaces are unused, but I think showing the list of used workspaces has wider applicability and it has enough information for me to discover the information I need.

I used to use Mod+Left and Mod+Right before I started using `xdotool`. It was difficult to press a key combo an unknown number of times to get back to the current one, while simultaneously trying to keep track of all the workspace numbers I had seen so that I could find one that I hadn't.

Right now, I'm using `xdotool` in my `bar_action` script to synthesise the list of used workspaces. There are two downsides, though neither of them are deal-breakers:

1. The bar does not update immediately. The more frequently my `bar_action` script updates the bar the more frequently it runs xdotool, when the vast majority of times absolutely nothing has changed. It's gratuitously inefficient to have it update very often.
2. The number list has to appear together with the rest of the information generated by the `bar_action`. I have a specific place I would like to put the number list, but it can't appear there without shuffling some other part of the bar around.

Now that I think about it, a better way to solve this problem would be for me to repeatedly press  Mod-Shift-Up until I arrived at an unused workspace...

...I may have just talked myself out of this feature request.
